packages/commons/src/events/Action.ts

```diff
@@ -62,18 +61,28 @@ export const ActionInputBase = z.object({
   )
 })
 
-export const ActionInput = z.union([
-  ActionInputBase.extend({
-    type: z.enum([ActionType.CREATE])
-  }),
-  ActionInputBase.extend({
-    type: z.enum([ActionType.REGISTER]),
-    identifiers: z.object({
-      trackingId: z.string(),
-      registrationNumber: z.string()
-    })
+export const CreateActionInput = ActionInputBase.extend({})
+export const NotifyActionInput = ActionInputBase.extend({})
+export const DeclareActionInput = ActionInputBase.extend({})
+export const RegisterActionInput = ActionInputBase.extend({
+  identifiers: z.object({
+    trackingId: z.string(),
+    registrationNumber: z.string()
   })
-])
+})
+
+export const ActionInput = z
+  .union([
+    CreateActionInput.extend({ type: z.enum([ActionType.CREATE]) }),
+    NotifyActionInput.extend({ type: z.enum([ActionType.NOTIFY]) }),
+    DeclareActionInput.extend({ type: z.enum([ActionType.DECLARE]) }),
+    RegisterActionInput.extend({ type: z.enum([ActionType.REGISTER]) })
+  ])
+  .and(
```

Comment:
Is there semantic difference between `extend` and `and`?

Reply:
.and apparently creates `{ name: string } & { age: number }` while .extend creates `{ name: string; age: number }`

https://zod.dev/?id=merge
vs
https://zod.dev/?id=and